src/query/v2/shard_request_manager.hpp

```diff
@@ -519,11 +536,22 @@ class ShardRequestManager : public ShardRequestManagerInterface {
   }
 
   void SendAllRequests(ExecutionState<ScanVerticesRequest> &state) {
+    int64_t shard_idx = 0;
     for (const auto &request : state.requests) {
-      auto &storage_client =
-          GetStorageClientForShard(*state.label, storage::conversions::ConvertPropertyVector(request.start_id.second));
-      ReadRequests req = request;
-      storage_client.SendAsyncReadRequest(request);
+      if (!state.label) {
+        const auto &current_shard = state.shard_cache[shard_idx];
+
+        auto &storage_client = GetStorageClientForShard(current_shard);
+        ReadRequests req = request;
+        storage_client.SendAsyncReadRequest(request);
+      } else {
+        auto &storage_client = GetStorageClientForShard(
+            *state.label, storage::conversions::ConvertPropertyVector(request.start_id.second));
```

Comment:
I propose to use the same way of getting the storage client because of multiple reasons:
1. It is faster
2. Relying on the `request.start_id` is really bad in my opinion as that field should become optional soon.
3. It makes the logic simple as it eliminates on if/else.